tests/conftest.py

```diff
@@ -14,13 +15,52 @@ def remove_headers(response: dict):
         response["headers"] = {}
         return response
 
+    def scrub_requests(request: dict):
+        if request.body:
+            try:
+                body_str = request.body.decode("utf-8")
+                body_json = json.loads(body_str)
+                if "token" in body_json:
+                    body_json["token"] = "ntn_..."
+                if "code" in body_json:
+                    body_json["code"] = "..."
+                if "redirect_uri" in body_json:
+                    body_json["redirect_uri"] = "http://..."
+                request.body = json.dumps(body_json).encode("utf-8")
+
+            except (json.JSONDecodeError, AttributeError):
+                pass
+        return request
+
+    def scrub_response(response: dict):
+        if "content" in response:
+            try:
+                content_json = json.loads(response["content"])
+                if "access_token" in content_json:
+                    response["content"] = json.dumps(
+                        {key: "..." for key in content_json}, separators=(",", ":")
+                    )
+
+            except json.JSONDecodeError:
+                pass
+
+        return response
+
+    def scrub_auth_header(key: str, value: str, request: Optional[object] | None):
```

Comment:
I think the type hint for `request` is wrong here, and likely to be the source of the CI errors: `Optional` already means "something or `None`", so this union probably doesn't make sense, and IIRC unions require the `Union` type for older Python versions.

Also I'm quite certain we can make this `object` type more precise. :)